mlflow/langchain/runnables.py

```diff
@@ -137,6 +142,92 @@ def runnable_sequence_from_steps(steps):
+            "in order to load runnable with steps."
+        )
+
+    branches_conf_file = load_path / _RUNNABLE_BRANCHES_FILE_NAME
+    if not branches_conf_file.exists():
+        raise MlflowException(
+            f"File {branches_conf_file} must exist in order to load runnable with steps."
+        )
+    branches_conf = _load_from_yaml(branches_conf_file)
+    branches_path = load_path / _BRANCHES_FOLDER_NAME
+    if not branches_path.exists() or not branches_path.is_dir():
+        raise MlflowException(
+            f"Folder {branches_path} must exist and must be a directory "
+            "in order to load runnable with steps."
+        )
+
+    branches = []
+    for branch in os.listdir(branches_path):
+        # load model from the folder of the branch
+        if branch == _DEFAULT_BRANCH_NAME:
+            default_branch_path = branches_path / _DEFAULT_BRANCH_NAME
+            default = _load_model_from_path(
+                default_branch_path, branches_conf.get(_DEFAULT_BRANCH_NAME)
+            )
+        else:
+            branch_tuple = []
+            for i in range(2):
+                config = branches_conf.get(f"{branch}-{i}")
+                runnable = _load_model_from_path(
+                    os.path.join(branches_path, branch, str(i)), config
+                )
+                branch_tuple.append(runnable)
+            branches.append(tuple(branch_tuple))
+
+    # default branch must be the last branch
+    branches.append(default)
+
+    return RunnableBranch(*branches)
+
+
+def _save_internal_runnables(runnable, path, loader_fn, persist_dir):
+    conf = {}
+    if isinstance(runnable, lc_runnables_types()):
+        conf[_MODEL_TYPE_KEY] = runnable.__class__.__name__
+        conf.update(_save_runnables(runnable, path, loader_fn, persist_dir))
+    elif isinstance(runnable, lc_runnable_branch_type()):
+        conf[_MODEL_TYPE_KEY] = runnable.__class__.__name__
+        conf.update(_save_runnable_branch(runnable, path, loader_fn, persist_dir))
+    elif isinstance(runnable, base_lc_types()):
+        lc_model = _validate_and_wrap_lc_model(runnable, loader_fn)
+        conf[_MODEL_TYPE_KEY] = lc_model.__class__.__name__
+        conf.update(_save_base_lcs(lc_model, path, loader_fn, persist_dir))
+    else:
+        conf = {
+            _MODEL_TYPE_KEY: runnable.__class__.__name__,
+            _MODEL_DATA_KEY: _MODEL_DATA_YAML_FILE_NAME,
+            _MODEL_LOAD_KEY: _CONFIG_LOAD_KEY,
+        }
+        path = path / _MODEL_DATA_YAML_FILE_NAME
+        # Save some simple runnables that langchain natively supports.
+        if hasattr(runnable, "save"):
+            runnable.save(path)
+        # TODO: check if `dict` is enough to load it back
+        elif hasattr(runnable, "dict"):
+            runnable_dict = runnable.dict()
+            with open(path, "w") as f:
+                yaml.dump(runnable_dict, f, default_flow_style=False)
+        else:
+            return
```

Comment:
Do we want a silent failure to serialize or should the user be alerted that they're saving an incomplete model?